packages/geolocator/tizen/src/geolocator_tizen_plugin.cc

```diff
@@ -90,111 +90,101 @@ class GeolocatorTizenPlugin : public flutter::Plugin {
     geolocator_updates_channel_->SetStreamHandler(std::move(handler));
   }
 
-  void HandleMethodCall(
-      const flutter::MethodCall<flutter::EncodableValue> &method_call,
-      std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result) {
+  void HandleMethodCall(const FlMethodCall &method_call,
+                        std::unique_ptr<FlMethodResult> result) {
     std::string method_name = method_call.method_name();
+
+    result_ = std::move(result);
+
     if (method_name == "checkPermission") {
-      OnCheckPermission(std::move(result));
+      OnCheckPermission();
     } else if (method_name == "isLocationServiceEnabled") {
-      OnIsLocationServiceEnabled(std::move(result));
+      OnIsLocationServiceEnabled();
     } else if (method_name == "requestPermission") {
-      OnRequestPermission(std::move(result));
+      OnRequestPermission();
     } else if (method_name == "getLastKnownPosition") {
-      OnGetLastKnownPosition(std::move(result));
+      OnGetLastKnownPosition();
     } else if (method_name == "getCurrentPosition") {
-      OnGetCurrentPosition(std::move(result));
+      OnGetCurrentPosition();
     } else if (method_name == "openAppSettings") {
       TizenResult ret = Setting::LaunchAppSetting();
-      result->Success(flutter::EncodableValue(static_cast<bool>(ret)));
+      SendResult(flutter::EncodableValue(static_cast<bool>(ret)));
     } else if (method_name == "openLocationSettings") {
       TizenResult ret = Setting::LaunchLocationSetting();
-      result->Success(flutter::EncodableValue(static_cast<bool>(ret)));
+      SendResult(flutter::EncodableValue(static_cast<bool>(ret)));
     } else {
       result->NotImplemented();
     }
   }
 
-  void OnCheckPermission(
-      std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result) {
+  void OnCheckPermission() {
     PermissionStatus permission_status;
     TizenResult ret =
         permission_manager_->CheckPermissionStatus(&permission_status);
     if (!ret) {
-      result->Error("Failed to check permssion status.", ret.message());
+      SendErrorResult(ret.message(), "Failed to check permssion status.");
       return;
     }
     LOG_INFO("permission_status is %d", permission_status);
-    result->Success(
-        flutter::EncodableValue(static_cast<int>(permission_status)));
+    SendResult(flutter::EncodableValue(static_cast<int>(permission_status)));
   }
 
-  void OnIsLocationServiceEnabled(
-      std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result) {
+  void OnIsLocationServiceEnabled() {
     bool is_enabled = false;
     TizenResult ret = location_manager_->IsLocationServiceEnabled(&is_enabled);
 
     // TODO : add location service listener
     if (!ret) {
-      result->Error("Failed to check service enabled.", ret.message());
+      SendErrorResult(ret.message(), "Failed to check service enabled.");
```

Comment:
Why using an error message as an error code? Since the Dart side caller can clearly know that the failed method name is `isLocationServiceEnabled`, you don't have to repeat the information in the error message and can just use this format:

```cpp
SendErrorResult(std::to_string(ret.error_code()), ret.message());
// or
SendErrorResult("Operation failed", ret.message());
```